File: compatLib/src/main/java/app/lawnchair/compatlib/RemoteTransitionCompat.java
package app.lawnchair.compatlib;

import android.app.IApplicationThread;
import android.window.IRemoteTransition;
import android.window.RemoteTransition;
import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

public interface RemoteTransitionCompat {

    RemoteTransition getRemoteTransition(
            @NonNull IRemoteTransition remoteTransition,
            @Nullable IApplicationThread appThread,
            @Nullable String debugName);
}

File: compatLib/src/app/lawnchair/compatlib/RemoteTransitionCompat.kt
package app.lawnchair.compatlib;

import android.annotation.NonNull;
import android.annotation.Nullable;
import android.app.IApplicationThread;
import android.window.IRemoteTransition;
import android.window.RemoteTransition;

abstract class RemoteTransitionCompat {

    abstract fun getRemoteTransition(
        @NonNull remoteTransition: IRemoteTransition,
        @Nullable appThread: IApplicationThread?,
        @Nullable debugName: String?
    ): RemoteTransition
}
